src/Inputs/Switch.tsx

```diff
@@ -1,133 +1,127 @@
 import React from 'react';
 import styled from 'styled-components';
 import { position, darken, flex, transition } from '../Utils/Mixins';
-import { LabelLayout, LabelLayoutProps } from '../Fragments';
 
-const COMPONENT = (
-    props = {},
-    innerProps = {},
-    value?: boolean,
-): React.ReactElement => (
-    <Container>
-        <Input
-            type="checkbox"
-            {...props}
-            value={value === undefined ? '' : value.toString()}
-            checked={value}
-        />
-        <SwitchBox {...innerProps}>
-            <SwitchDot />
-        </SwitchBox>
-    </Container>
-);
 
-const withTags = (
-    props: object,
-    innerProps: {
-        activeStyle?: Function;
-        switchStyle?: Function;
-    },
-    tags?: string[],
-    value?: boolean,
-): React.ReactElement => {
-    if (tags) {
-        return (
-            <Tags>
-                {tags[0] && <Tag>{tags[0]}</Tag>}
-                {COMPONENT(props, innerProps, value)}
-                {tags[1] && <Tag>{tags[1]}</Tag>}
-            </Tags>
-        );
-    }
-
-    return COMPONENT(props, innerProps, value);
-};
+export interface SwitchProps extends InputProps {
+    leftTag?:string;
+    rightTag?:string;
+    activeColor?:string;
+    switchColor?:string;
+    label?:string;
+    description?:string;
+}
 
-export interface SwitchProps extends LabelLayoutProps {
-    tags?: string[];
-    activeStyle?: Function;
-    switchStyle?: Function;
-    value?: boolean;
+interface InputProps {
+    isChecked?: boolean;
+    onChange?: Function;
+    disabled?: boolean;
 }
 
 export const Switch: React.FC<SwitchProps> = ({
-    tags,
-    activeStyle,
-    switchStyle,
-    value,
-    ...props
-}): React.ReactElement => {
-    const component = withTags(
-        props,
-        { activeStyle, switchStyle },
-        tags,
-        value,
-    );
-    return <LabelLayout {...props}>{component}</LabelLayout>;
-};
+    leftTag,
+    rightTag,
+    activeColor,
+    switchColor,
+    onChange = (): void => {},
+    isChecked,
+    label,
+    description,
+    disabled
+}): React.ReactElement => (
+    <Container>
+        <Layout>
+            <Label>{label}</Label>
+            <Info>{description}</Info>
+        </Layout>
+        <Row>
+            {leftTag && <Tag margin="5px 10px 0 0">{leftTag}</Tag>}
+            <Container>
+                <Input
+                    type="checkbox"
+                    onChange={(e): void => onChange(e)}
```

Comment:
only issue i have with this is that it wouldnt be .value, but .checked which would have to be implemented everywhere else